File: Singleton/Singleton.ts
class Singleton {
  private static _instance: Singleton | null = null;

  private constructor() { }

  public static getInstance(): Singleton {
    if (!Singleton._instance) {
      Singleton._instance = new Singleton();
    }
    return Singleton._instance;
  }

  public showMessage(): void {
    console.log("Exemplo de Singleton em TypeScript");
  }
}

File: Singleton/Singleton.cs
using System;

public sealed class Singleton
{
  private static Singleton _instance = null;
  private static readonly object _padlock = new object();

  private Singleton() { }

  public static Singleton Instance
  {
    get
    {
      lock (_padlock)
      {
        return _instance ??= new Singleton();
      }
    }
  }

  public void ShowMessage()
  {
    Console.WriteLine("Exemplo de Singleton em C#");
  }
}
